tests/execution_engine/test_sparkdf_execution_engine.py

```diff
@@ -184,6 +188,153 @@ def test_get_domain_records_with_column_domain(
     ), "Data does not match after getting full access compute domain"
 
 
+def test_get_domain_records_with_column_domain_and_filter_conditions(
+    spark_session, basic_spark_df_execution_engine
+):
+    pd_df = pd.DataFrame(
+        {"a": [1, 2, 3, 4, 5], "b": [2, 3, 4, 5, None], "c": [1, 2, 3, 4, None]}
+    )
+    df = spark_session.createDataFrame(
+        [
+            tuple(
+                None if isinstance(x, (float, int)) and np.isnan(x) else x
+                for x in record.tolist()
+            )
+            for record in pd_df.to_records(index=False)
+        ],
+        pd_df.columns.tolist(),
+    )
+    engine = basic_spark_df_execution_engine
+    engine.load_batch_data(batch_id="1234", batch_data=df)
+    data = engine.get_domain_records(
+        domain_kwargs={
+            "column": "a",
+            "row_condition": 'col("b")<5',
+            "condition_parser": "great_expectations__experimental__",
+            "filter_conditions": [
```

Comment:
@anthonyburdi I can see this programmatic way of specifying the filter conditions; how would a user specify them?  Is there a text/dictionary-only syntax?  Thanks.

Reply:
Great question! These are only specified via using the engine's `add_column_row_condition` method, which is accessed via expectation params or set to defaults internally. A user cannot specify them using yaml for example, the interface for user input row conditions is the `row_conditions` and `condition_parser` parameters.